api/internal/localizer/localizer.go

```diff
@@ -76,8 +85,62 @@ func (lc *Localizer) Localize() error {
 }
 
 // TODO(annasong): implement
-// processKust returns a copy of the kustomization at kt with paths localized.
-func (lc *Localizer) processKust(kt *target.KustTarget) *types.Kustomization {
-	kust := kt.Kustomization()
-	return &kust
+// localizeFields localizes paths on the kustomization's built-in fields, excluding helm.
+func (lc *Localizer) localizeFields(_ *types.Kustomization) error {
+	return nil
+}
+
+// localizePlugins localizes built-in plugins on kust that can contain file paths. The built-in plugins
+// can be inline or a file, but not the result of a kustomization.
+func (lc *Localizer) localizePlugins(kust *types.Kustomization) error {
+	for field, plugins := range map[string][]string{
+		"generator":   kust.Generators,
+		"transformer": kust.Transformers,
+		"validator":   kust.Validators,
+	} {
+		for i, entry := range plugins {
+			var isPath bool
+			rm, err := lc.rFactory.NewResMapFromBytes([]byte(entry))
+			if err != nil {
+				rm, err = lc.rFactory.FromFile(lc.ldr, entry)
+				if err != nil {
+					return errors.WrapPrefixf(err, "%q is neither a valid inline nor file path %s", entry, field)
```

Comment:
Thanks for this recommendation! Applied slight variant. Let me know if it needs to be further tweaked.

Reply:
What do the reviewers think about creating a new `error` here? Presumably we want to wrap the 2 errors so that upper layers can unwrap them. Since we can't wrap 2 errors, we can create a new error that upper layers can check for instead?